compiler-core/src/type_/expression.rs

```diff
@@ -2307,27 +2366,88 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
                     ValueConstructorVariant::Record { .. } => unreachable!(),
                 }
             }
-        }?;
+
+            Constant::Invalid { .. } => panic!("invalid constants can not be in an untyped ast"),
+        }
+    }
+
+    pub fn infer_const(
```

Comment:
This function has become quite hard to read and features a very large amount of nesting. How could we restructure it to be easier to follow?

Reply:
Thank you